packages/@aws-cdk/aws-iam/lib/role.ts

```diff
@@ -237,13 +245,10 @@ export class Role extends Resource implements IRole {
       }
     }
 
-    const roleAccount = parsedArn.account;
-
-    const scopeAccount = scopeStack.account;
-
-    return options.mutable !== false && accountsAreEqualOrOneIsUnresolved(scopeAccount, roleAccount)
-      ? new Import(scope, id)
-      : new ImmutableRole(scope, `ImmutableRole${id}`, new Import(scope, id));
+    const importedRole = new Import(scope, id);
+    return options.mutable !== false && accountsAreEqualOrOneIsUnresolved(scopeStack.account, importedRole.account)
```

Comment:
Can you spend a couple of comment lines here to explain why we have the account comparison here again? I keep on forgetting... :(

Reply:
Done.